splink/em_training_session.py

```diff
@@ -165,6 +166,20 @@ def _train(self):
 
         cvv = self._comparison_vectors()
 
+        # check that the blocking rule actually generates _some_ record pairs,
+        # if not give the user a helpful message
+        if not cvv.as_record_dict(limit=1):
+            br_sql = f"`{self._blocking_rule_for_training.blocking_rule}`"
+            raise EMTrainingException(
+                f"Training rule {br_sql} resulted in no record pairs. "
+                "This means that in the supplied data set "
+                f"there were no pairs of records for which {br_sql} was `true`.\n"
+                "Expectation maximisation requires a set of "
+                "record comparisons to work on. "
+                "You must revise your training blocking rule so that "
+                "the set of generated comparisons is not empty."
```

Comment:
Maybe worth being a little more specific with this wording.  Suggest something like:

```
Expectation maximisation requires a substantial number of record comparisons to produce accurate parameter estimates - usually at least a few hundred, but preferably a few thousand.
```

```
You must revise your training blocking rule so that the set of generated comparisons is not empty.  You can use `linker.count_num_comparisons_from_blocking_rule()` to compute the number of comparisons that will be generated by a blocking rule.
```